app/javascript/gobierto_data/webapp/components/Dataset.vue

```diff
@@ -809,43 +828,45 @@ export default {
       //Changes the path depending on if we save a query or viz.
       const pathQueryOrViz = this.savingViz ? 'v' : 'q'
 
-      /*Don't updates the URL if the component is Editor, only replace and don't reload, because in the editor we've two options, saved a query or viz, if the user saves a viz, and we update the URL, the browser reloads, and the user goes to visualization tab, and this behavior is too hacky.*/
-      //https://developer.mozilla.org/en-US/docs/Web/API/History/pushState
-      history.pushState(
-        {},
-        null,
-        `${location.origin}/datos/${slugDataset}/${pathQueryOrViz}/${newId}`
-      )
+      this.$router.push(`/datos/${slugDataset}/${pathQueryOrViz}/${newId}`)
 
       this.enabledForkButton = false
       this.queryInputFocus = false
+      this.enabledForkVizButton = false
     },
     getColumnsQuery(csv = '') {
       const [ columns = '' ] = csv.split("\n");
       this.arrayColumnsQuery = columns.split(",");
     },
-    resetQuery(value) {
-      this.resetQueryDefault = value
-      if (value === true) {
-        this.isQuerySavingPromptVisible = false
-        this.currentQuery = `SELECT * FROM ${this.tableName} LIMIT 50`;
-        this.isQueryModified = false
-        this.runCurrentQuery()
-        this.disabledSavedButton()
-        this.disabledStringSavedQuery()
-        this.queryName = null
-      }
+    resetQuery() {
+      this.isQuerySavingPromptVisible = false
+      this.currentQuery = `SELECT * FROM ${this.tableName} LIMIT 50`;
```

Comment:
You might want to create a computed prop (initialQuery) and replace it in both places